src/aliceVision/numeric/numeric.hpp

```diff
@@ -570,6 +570,20 @@ void SplitRange(const T range_start, const T range_end, const int nb_split,
   }
 }
 
+template<class T>
+constexpr T divideRoundUp(T x, T y)
+{
+    static_assert(std::is_integral<T>::value, "divideRoundUp only works with integer arguments");
+    auto xPos = x >= 0;
+    auto yPos = y >= 0;
+    if (xPos == yPos) {
+        return x / y + ((x % y) != 0 ? 1 : 0);
+    } else {
+        // negative result, rounds towards zero anyways
+        return x / y;
+    }
```

Comment:
I don't want to overthink it, I wonder if `std::div` could be more efficient
```suggestion
    if (xPos == yPos) {
        const auto dv = std::div(x, y);
        return dv.quot + (dv.rem != 0 ? 1 : 0);
    } else {
        // negative result, rounds towards zero anyways
        return x / y;
    }
```

Reply:
No, std::div always results in a function call. See e.g. https://godbolt.org/z/aKjrnYr6s.

Reply:
The current code is pretty much as efficient as it could get. The `if (xPos == yPos)` will be predicted and thus essentially free unless we start mixing in negative numbers. The remainder (if not provided by div instruction) can be computed by relatively inexpensive multiplication and subtraction. `(dv.rem != 0 ? 1 : 0)` will most likely result in conditional move, thus free. Essentially the cost of this function is pretty much a single division operation.

Reply:
Thanks for the explanation!
We can merge it